feature-pay-impl/src/main/java/io/novafoundation/nova/feature_pay_impl/presentation/shop/ShopFragment.kt

```diff
@@ -27,8 +76,35 @@ class ShopFragment : BaseFragment<ShopViewModel, FragmentShopBinding>() {
     }
 
     override fun subscribe(viewModel: ShopViewModel) {
-        viewModel.isWalletAvailableFlow.observe {
-            unavailableWalletAdapter.show(!it)
+        viewModel.brandsListState.observe {
+            val showUnavailableWalletCase = it is BrandsListState.UnavailableWallet
+            val showBrands = it is BrandsListState.Brands
+            val showLoading = it is BrandsListState.Brands && it.brands.isLoading
+
+            unavailableWalletAdapter.show(showUnavailableWalletCase)
+
+            headerAdapter.show(showBrands)
+            searchAdapter.show(showBrands)
+            shopPaginationLoadingAdapter.show(showBrands && !showLoading)
+            headerAdapter.showShimmering(showLoading)
+            shimmeringAdapter.show(showLoading)
+            brandsAdapter.submitList(it.getBrandsOrNull())
         }
+
+        viewModel.maxCashback.observe(headerAdapter::setHeaderText)
+
+        viewModel.isLastPageReached.observe { shopPaginationLoadingAdapter.setInvisible(it) }
```

Comment:
I think this logic is wrong:

1. We should not create a separate side effect for "last page loaded" because this information is already available via state: `state is FullData`
2. shopPaginationLoadingAdapter should be visible when the state is NewPageLoading. For other states it should be hidden